lib/std/event/loop.zig

```diff
@@ -512,21 +512,28 @@ pub const Loop = struct {
             },
             .kev = undefined,
         };
-        defer self.bsdRemoveKev(ident, filter);
+
+        defer {
+            // If the kevent was set to be ONESHOT, it doesn't need to be deleted manually.
+            if (flags & os.EV_ONESHOT != 0) {
```

Comment:
Do we maybe want to assert that flags always contains `EV_ONESHOT` and document the requirement in doc comments? It's true in all the callsites of this function.